test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/VerifyCommandTests.cs

```diff
@@ -280,5 +282,611 @@ public async Task VerifyCommand_VerifyOnPackageSignedWithoutAllowedCertificateFa
+            using (var zipStream = await package.CreateAsStreamAsync())
+            {
+                var packagePath = Path.Combine(pathContext.PackageSource, "testpackage.nupkg");
+
+                zipStream.Seek(offset: 0, loc: SeekOrigin.Begin);
+
+                using (var fileStream = File.OpenWrite(packagePath))
+                {
+                    zipStream.CopyTo(fileStream);
+                }
+
+                var signResult = CommandRunner.Run(
+                    _nugetExePath,
+                    pathContext.PackageSource,
+                    $"sign {packagePath} -CertificateFingerprint {cert.Source.Cert.Thumbprint} -CertificateStoreName {cert.StoreName} -CertificateStoreLocation {cert.StoreLocation}",
+                    waitForExit: true);
+
+                signResult.Success.Should().BeTrue();
+
+                var certificateFingerprintString = SignatureTestUtility.GetFingerprint(cert.Source.Cert, HashAlgorithmName.SHA256);
+
+                var trustedSignersSectionContent = $@"
+    <trustedSigners>
+        <author name=""signed"">
+            <certificate fingerprint=""{certificateFingerprintString}"" hashAlgorithm=""SHA256"" allowUntrustedRoot=""{trust}"" />
+        </author>
+    </trustedSigners>
+";
+                SimpleTestSettingsContext.AddSectionIntoNuGetConfig(pathContext.WorkingDirectory, trustedSignersSectionContent, "configuration");
+                string fingerprint = fingerPrintOption ? $"-CertificateFingerprint {certificateFingerprintString};abc;def" : string.Empty;
+
+                // Act
+                var verifyResult = CommandRunner.Run(
+                    _nugetExePath,
+                    pathContext.PackageSource,
+                    $"verify {packagePath} -Signatures {fingerprint}",
+                    waitForExit: true);
+
+                // Assert
+                // For certificate with trusted root setting allowUntrustedRoot to true/false doesn't matter
+                verifyResult.Success.Should().BeTrue();
+                verifyResult.AllOutput.Should().Contain(_noTimestamperWarningCode);
+            }
+        }
+
+        [CIOnlyTheory]
+        [InlineData("true", true)]
+        [InlineData("true", false)]
+        [InlineData("false", true)]
+        [InlineData("false", false)]
+        public async Task Verify_AuthorSignedPackageWithTrustedCertificate_RepositoryTag_Fails(string trust, bool fingerPrintOption)
+        {
+            // Arrange
+            TrustedTestCert<TestCertificate> cert = _testFixture.TrustedTestCertificateChain.Leaf;
+            var package = new SimpleTestPackageContext();
+
+            using (var pathContext = new SimpleTestPathContext())
+            using (var zipStream = await package.CreateAsStreamAsync())
+            {
+                var packagePath = Path.Combine(pathContext.PackageSource, "testpackage.nupkg");
+
+                zipStream.Seek(offset: 0, loc: SeekOrigin.Begin);
```

Comment:
Aren't there shorter create methods for SimpleTestPackageContext? 

Or is there another purpsoe as to why we're using this method.

Reply:
Sure, I addressed it. Usually, I copy existing test to make new one then got this part, I should have addressed earlier.

Reply:
We have a lot tests that are overly verbose, which makes it hard to get to the point of the tests, because we have so many lines focusing on setup.